ChoosePython.cmake

```diff
@@ -0,0 +1,86 @@
+##
+## Molio
+##
+## Copyright (c) 2012-2015 Cajal Blue Brain, BBP/EPFL
+## All rights reserved. Do not distribute without permission.
+##
+## Responsible Author: Juan Hernando Vieites (JHV)
+## contact: jhernando@fi.upm.es
+
+# This script provides an option to choose which Python version should be
+# requested to FindPythonInterp and FindPythonLibs. The user can choose
+# between version 2, 3 or autodetection (the default). If
+# CHOOSE_PYTHON_REQUIRE_BOOST is set, autodetection will choose Python 3 only
+# if Boost.Python is found to have support for Python 3.
+#
+# Input Variables:
+# * CHOOSE_PYTHON_REQUIRE_BOOST
+#
+# Output Variables:
+# * ${PROJECT_NAME}_USE_PYTHON3: Set to 1 if and only if Python 3 is chosen.
+# * ${PROJECT_NAME}_USE_BOOST_PYTHON_VERSION: Equals 3 if Python 3 is choosen,
+#     empty string otherwise
+# * PYTHON_LIBRARY_SUFFIX: The suffix path where Python site packages are
+#     to be installed for the chosen Python version.
+# * PythonLibs_FIND_VERSION
+#   PythonInter_FIND_VERSION: Variables used by Python find scripts as input.
+#     Set to 2 or 3 depending on the version chosen.
+#
+# Input/output varaibles:
+# * USE_PYTHON_VERSION: (input/output), the Python version chosen. At return it
+#   can only be 2 or 3.
+#
+# Defines:
+# * -D${PROJECT_NAME}_USE_PYTHON3=1: If Python 3 is chosen.
+
+include(FindBoostConfig) # Including the workarounds for boost finders.
+
+set(USE_PYTHON_VERSION "auto" CACHE STRING
+  "Choose the required Python version (2, 3 or auto).")
+
+set_property(CACHE USE_PYTHON_VERSION PROPERTY STRINGS 2 3 auto)
+
+if(NOT PYTHONINTERP_FOUND AND ${USE_PYTHON_VERSION} STREQUAL auto)
+  find_package(PythonInterp 3 QUIET)
+  if(CHOOSE_PYTHON_REQUIRE_BOOST)
+    find_package(Boost COMPONENTS python3 QUIET)
+  endif()
+
+  if(PYTHONINTERP_FOUND AND (NOT CHOOSE_PYTHON_REQUIRE_BOOST OR Boost_FOUND))
+    set(USE_PYTHON_VERSION 3)
```

Comment:
It doesn't, but It's not meant to rewrite the cache value (actually I don't care whether it does or not), but to pass the detected version downstream.